goflags.go

```diff
@@ -23,31 +23,43 @@ type FlagSet struct {
 	Marshal     bool
 	description string
 	flagKeys    InsertionOrderedMap
+	groups      []groupData
 }
 
-type flagData struct {
+type groupData struct {
+	name string
+	description string
+}
+
+type FlagData struct {
 	usage        string
 	short        string
 	long         string
+	group        string // unused unless set later
 	defaultValue interface{}
 }
 
+// Group sets the group for a flag data
+func (f *FlagData) Group(name string) {
+	f.group = name
+}
+
 // NewFlagSet creates a new flagSet structure for the application
 func NewFlagSet() *FlagSet {
 	return &FlagSet{flagKeys: *newInsertionOrderedMap()}
 }
 
 func newInsertionOrderedMap() *InsertionOrderedMap {
 	return &InsertionOrderedMap{
-		values: make(map[string]*flagData),
+		values: make(map[string]*FlagData),
 		keys:   make([]string, 0, 0),
 	}
 }
 
 // Hash returns the unique hash for a flagData structure
 // NOTE: Hash panics when the structure cannot be hashed.
-func (flagSet *flagData) Hash() string {
-	hash, _ := structhash.Hash(flagSet, 1)
+func (f *FlagData) Hash() string {
```

Comment:
From what I've read, and also what the IDE suggests, the receiver names should be uniform throughout the file. Even though the "official" recommendation is to keep the variable names short (which I don't agree with, except it's holding an index, or something that has a very short life), I prefer variables that clearly describe the value they hold, especially if the method gets longer, and does not fit in one page. Scrolling back and forth to understand cryptic variable names is annoying.

Reply:
Okay, will make it uniform with the rest.